addon/components/impagination-collection.js

```diff
@@ -44,11 +38,15 @@ export default Ember.Component.extend({
     this.set('datasetState', this.get('dataset.state'));
   }),
 
+  safeSet: function(key, value) {
+    if (!this.isDestroyed) { this.set(key, value); }
+  },
```

Comment:
yup that's exactly true. I did this on the `observe` method which is passes to `impagination`, called asynchronously, and then we change the state of our component.